app/components/LoginPage.tsx

```diff
@@ -23,44 +65,44 @@ export default class LoginPage extends Component {
 
     onSubmit = () => {
         const { password, username } = this.state;
-        const { isSamlEnabled, location, onLogin, samlSsoUrl } = this.props;
-        const errors = {};
+        const { isSamlEnabled, location, onLogin, samlSsoUrl = '' } = this.props;
+        const errors: Errors = {};
 
         if (isSamlEnabled) {
-            // eslint-disable-next-line scanjs-rules/assign_to_location
-            window.location = samlSsoUrl;
+            // eslint-disable-next-line scanjs-rules/assign_to_href
+            window.location.href = samlSsoUrl;
         }
 
         if (_.isEmpty(username)) {
-            errors.username = i18n.t('login.error.noUsername', 'Please provide username');
+            errors.username = t('error.noUsername');
         }
         if (_.isEmpty(password)) {
-            errors.password = i18n.t('login.error.noPassword', 'Please provide password');
+            errors.password = t('error.noPassword');
         }
         if (!_.isEmpty(errors)) {
             this.setState({ errors });
             return false;
         }
 
         const query = parse(location.search);
-        const redirect = query.redirect || null;
+        const redirect = (query.redirect as string) || null;
 
         return onLogin(username, password, redirect);
     };
 
-    handleInputChange = (proxy, field) => {
+    handleInputChange = (_proxy: any, field: Parameters<typeof Stage.Basic.Form.fieldNameValue>[0]) => {
```

Comment:
As `Form` should be available here, we can use shorter form:
```suggestion
    handleInputChange = (_proxy: any, field: Parameters<typeof Form.fieldNameValue>[0]) => {
```